lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,18 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(@parameters[:name].value) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      file_resources = catalog.resources.reject { |resource| resource.type != Puppet::Type.type(:file) }
+      children_file_resources = file_resources.select { |resource| resource.parameters[:path].value =~ %r(#{@parameters[:name].value}/.) }
```

Comment:
You lost me a little here. I think that is probably because I am a polyglot and tend to just look at languages sometimes as doing C++ in that language (hence why I was googling 'ruby copy constructor' yesterday).

Basically my interpretation of what you are saying is I should be accessing the instance of `Puppet::Type::File/Mount` and then the hash value of its parameter from there, instead of an instance of `Puppet::Type::File/Mount::Parameter` and accessing its instance variable from there.

As for `@parameters[:name].value` versus `self[:name]`, I was just doing a quick paste from the autorequire block for nested mounts since I know that was accurately accessing the mount name parameter. I will try changing the autorequire block I added and see what happens since it seems that part was passing my new spec tests.